core/trino-main/src/main/java/io/trino/server/Server.java

```diff
@@ -268,12 +269,10 @@ private static void updateConnectorIds(Announcer announcer, CatalogManager catal
 
     private static ServiceAnnouncement getTrinoAnnouncement(Set<ServiceAnnouncement> announcements)
     {
-        for (ServiceAnnouncement announcement : announcements) {
-            if (announcement.getType().equals("trino")) {
-                return announcement;
-            }
-        }
-        throw new IllegalArgumentException("Trino announcement not found: " + announcements);
+        return announcements.stream()
+                .filter(announcement -> announcement.getType().equals("trino"))
+                .collect(toOptional())
```

Comment:
will it produce readable error if there is more than one occurence?

Reply:
not very awesome message, but that's rather impossible to happen, so i wouldn't be worried